File: workoutpixel/src/main/java/ch/karimattia/workoutpixel/database/AppDatabase.java
package ch.karimattia.workoutpixel.database;

import android.content.Context;

import androidx.room.Database;
import androidx.room.Room;
import androidx.room.RoomDatabase;

import ch.karimattia.workoutpixel.core.Goal;

@Database(entities = {PastWorkout.class, Goal.class}, version = 2)
public abstract class AppDatabase extends RoomDatabase {
    // private static final Object sLock = new Object();
    private static AppDatabase workoutPixelDb;

    // Return database - if there already exists one, return this one, otherwise create one.
    public static AppDatabase getDatabase(Context context) {
        // synchronized (sLock) {
            if (null == workoutPixelDb) {
                workoutPixelDb = buildDatabaseInstance(context);
            }
            return workoutPixelDb;
        }
    // }

    private static AppDatabase buildDatabaseInstance(Context context) {
        return Room.databaseBuilder(context,
                AppDatabase.class,
                "WorkoutPixelDatabase")
                .allowMainThreadQueries()
                //.addMigrations(MIGRATION_1_2)
                //.fallbackToDestructiveMigration()
                .build();
    }

    public abstract WorkoutDao workoutDao();

// --Commented out by Inspection START (23.06.21, 20:29):
//    static final Migration MIGRATION_1_2 = new Migration(1, 2) {
//        @Override
//        public void migrate(SupportSQLiteDatabase database) {
//            database.execSQL("ALTER TABLE pastWorkouts ADD widgetUid int;");
//        }
//    };
// --Commented out by Inspection STOP (23.06.21, 20:29)

}

File: workoutpixel/src/main/java/ch/karimattia/workoutpixel/data/AppDatabase.kt
package ch.karimattia.workoutpixel.data

import android.content.Context
import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.android.qualifiers.ApplicationContext
import dagger.hilt.components.SingletonComponent
import javax.inject.Singleton

@Database(entities = [PastWorkout::class, Goal::class], version = 2)
abstract class AppDatabaseKotlin : RoomDatabase() {
	abstract fun goalDao(): GoalDao
}

@InstallIn(SingletonComponent::class)
@Module
class DatabaseModule {

	@Provides
	fun provideGoalDao(appDatabaseKotlin: AppDatabaseKotlin): GoalDao {
		return appDatabaseKotlin.goalDao()
	}

	@Provides
	@Singleton
	fun provideAppDatabase(@ApplicationContext context: Context): AppDatabaseKotlin {
		return Room.databaseBuilder(
			context,
			AppDatabaseKotlin::class.java,
			"WorkoutPixelDatabase"
		)
			.allowMainThreadQueries()
			//.addMigrations(MIGRATION_1_2)
			//.fallbackToDestructiveMigration()
			.build()
	}
}